submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepository.kt

```diff
@@ -45,6 +50,23 @@ class SubmissionDocDataRepository(
     private val submissionRepository: SubmissionMongoRepository,
     private val mongoTemplate: ReactiveMongoTemplate,
 ) : SubmissionMongoRepository by submissionRepository {
+
+    suspend fun saveSubmission(docSubmission: DocSubmission): DocSubmission {
+        return submissionRepository.save(docSubmission).awaitSingle()
+    }
+
+    suspend fun saveAllSubmissions(submissions: List<DocSubmission>) {
+        submissionRepository.saveAll(submissions).awaitFirst()
```

Comment:
I'm wondering, if this is an operation to save all the elements, shouldn't we use `awaitLast()` instead to make sure all of them were processed? Or it doesn't make a difference?

Reply:
make sense yes, great catch